src/__tests__/organization.tests.ts

```diff
@@ -0,0 +1,319 @@
+import { Utility } from "../utils";
+import { AuthApi, Register, RoleDetails, Roles, User, UserManagementApi, OrganizationApi, Organization, OrganizationList } from "tdei-management-client";
+import { faker } from '@faker-js/faker';
+import { it } from "node:test";
+import seed, { SeedDetails } from "../data.seed";
+import { TdeiObjectFaker } from "../tdei-object-faker";
+import exp from "node:constants";
+
+describe("Organization service", () => {
+  let configurationWithAuthHeader = Utility.getConfiguration();
+  let configurationWithoutAuthHeader = Utility.getConfiguration();
+  let seederData: SeedDetails | undefined = undefined;
+  beforeAll(async () => {
+    seederData = await seed.generate();
+    let generalAPI = new AuthApi(configurationWithAuthHeader);
+    const loginResponse = await generalAPI.authenticate({
+      username: configurationWithAuthHeader.username,
+      password: configurationWithAuthHeader.password
+    });
+    configurationWithAuthHeader.baseOptions = {
+      headers: { ...Utility.addAuthZHeader(loginResponse.data.access_token) }
+    };
+  });
+
+  describe("Create Organization", () => {
+    describe("Auth", () => {
+      it("When no auth token provided, Expect to return HTTP status 401", async () => {
+        //Arrange
+        let oraganizationApi = new OrganizationApi(configurationWithoutAuthHeader);
+        //Act
+        const request = async () => {
+          await oraganizationApi.createOrganization(<Organization>{})
+        }
+        //Assert
+        expect(request()).rejects.toMatchObject({response: { status: 401 } })
+      });
+    });
+
+    describe("Functional", () => {
+    it("When creating new oraganization, Expect to return newly created org id", async () => {
+      //Arrange
+      let oraganizationApi = new OrganizationApi(configurationWithAuthHeader);
+      //Act
+      const organizationResponse = await oraganizationApi.createOrganization(TdeiObjectFaker.getOrganization());
+      //Assert
+      expect(organizationResponse.status).toBe(200);
+      expect(organizationResponse.data.data?.length).toBeGreaterThan(0);
+
+    });
+
+    it("When creating new oraganization with same name, Expect to return HTTP Status 400", async () => {
+      //Arrange
+      let oraganizationApi = new OrganizationApi(configurationWithAuthHeader);
+      //Act
+      let payload = TdeiObjectFaker.getOrganization();
+      //TODO: correct it to name instead of id
+      payload.org_name = <string>seederData?.organizationId
```

Comment:
Please look at the implementation in pathways for similar case

Reply:
fixed